keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,633 @@
+            filters=num_classes,
+            kernel_size=1,
+            name=f"{cur_name}_class_3_conv",
+        )(class_predictions)
+        class_predictions = layers.Activation(
+            "sigmoid", name=f"{cur_name}_classifier"
+        )(class_predictions)
+
+        out = tf.concat([box_predictions, class_predictions], axis=-1)
+        out = layers.Reshape(
+            [-1, out.shape[-1]], name=f"{cur_name}_output_reshape"
+        )(out)
+        outputs.append(out)
+
+    outputs = tf.concat(outputs, axis=1)
+    outputs = layers.Activation("linear", dtype="float32", name="box_outputs")(
+        outputs
+    )
+
+    return {
+        "boxes": outputs[:, :, :BOX_REGRESSION_CHANNELS],
+        "classes": outputs[:, :, BOX_REGRESSION_CHANNELS:],
+    }
+
+
+def decode_regression_to_boxes(preds):
+    """
+    Decodes the results of the YOLOV8Detector forward-pass into left / top /
+    right / bottom predictions with respect to anchor boxes.
+
+    Each coordinate is encoded with 16 predicted values. Those predictions are
+    softmaxed and multiplied by [0..15] to make predictions. The resulting
+    predictions are relative to the stride of an anchor box (and correspondingly
+    relative to the scale of the feature map from which the predictions came).
+    """
+    preds_bbox = tf.reshape(
+        preds, (-1, preds.shape[1], 4, BOX_REGRESSION_CHANNELS // 4)
+    )
+    preds_bbox = tf.nn.softmax(preds_bbox, axis=-1) * tf.range(
+        BOX_REGRESSION_CHANNELS // 4, dtype="float32"
+    )
+    return tf.reduce_sum(preds_bbox, axis=-1)
+
+
+def dist2bbox(distance, anchor_points):
+    """
+    Decodes left / top / right / bottom predictions into xyxy box predictions.
+    The resulting xyxy predictions must be scaled by the stride of their
+    corresponding anchor points to yield an absolute xyxy box.
+    """
+    left_top, right_bottom = tf.split(distance, 2, axis=-1)
+    x1y1 = anchor_points - left_top
+    x2y2 = anchor_points + right_bottom
+    return tf.concat((x1y1, x2y2), axis=-1)  # xyxy bbox
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class YOLOV8Detector(Task):
+    """
+    Implements the YOLOV8 architecture for object detection.
+
+    Args:
+        num_classes: integer, the number of classes in your dataset excluding the
+            background class. Classes should be represented by integers in the
+            range [0, num_classes).
+        bounding_box_format: string, the format of bounding boxes of input dataset.
+            Refer
+            [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+            for more details on supported bounding box formats.
+        backbone: `keras.Model`, must implement the `pyramid_level_inputs`
+            property with keys 2, 3, and 4 and layer names as values. A
+            sensible backbone to use is the `keras_cv.models.YOLOV8Backbone`.
+        fpn_depth: integer, a specification of the depth for the Feature
```

Comment:
This arg description was better on the `apply_fpn` function. Could you reuse that here?

When reading this I originally thought it referred to how many P-levels are used in the FPN

Reply:
Ahh good catch -- done.